client/Containers/PaginationControl.jsx

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import styled from 'styled-components';
+import productPagination, {
+  setProductPagination,
+  nextPage,
+  prevPage,
+  goToPage,
+  sizePage,
+  sortPage,
+} from '../store/product/productPagination';
+import buildProductQuery from '../components/utils/buildProductQuery';
+import { fetchProductList } from '../store/product/productList';
+
+// const mapStateToProps = ({ productPagination }) => ({ productPagination });
+
+// const mapDispatchToProps = (dispatch) => ({
+//   setPagination: (productPagination) =>
+//     dispatch(setProductPagination(productPagination)),
+//   paginationFuncs: {
+//     nextPage: () => dispatch(nextPage()),
+//     prevPage: () => dispatch(prevPage()),
+//     goToPage: (page) => dispatch(goToPage(page)),
+//     sizePage: (size) => {
+//       dispatch(sizePage(size));
+//       dispatch(goToPage(1));
+//     },
+//     sortPage: (sort) => dispatch(sortPage(sort)),
+//   },
+// });
+
+const PaginationBar = styled.div`
+  display: flex;
+  justify-content: space-between;
+  background: black;
+  width: calc(100%);
+  border-radius: ${(props) => (props.top ? '0.5rem 0.5rem 0 0' : '0.5rem')};
+`;
+
+const PageOptionContainer = styled.div`
+  display: flex;
+`;
+
+const PageOption = styled.div`
+  padding: 1em;
+  color: ${(props) =>
+    props.operable ? 'white' : props.current ? '#ffda08' : 'gray'};
+  cursor: ${(props) => (props.operable ? 'pointer' : 'default')};
+`;
+
+const PaginationControl = ({ top }) => {
+  const { productFilters, productPagination } = useSelector(
+    ({ productFilters, productPagination }) => ({
+      productFilters,
+      productPagination,
+    })
+  );
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const getProducts = (query) => dispatch(fetchProductList(query));
+  const paginate = (options) => {
+    // options: { page?: INT, size?: INT, sort?: STRING }
+    for (const key in productPagination) {
+      if (options[key]) productPagination[key] = options[key];
+    }
+    const query = buildProductQuery({ productFilters, productPagination });
+    history.push(`/products${query}`);
+    getProducts(query);
+  };
```

Comment:
As above, will remove `dispatch` and `getProducts` as they are not necessary here